### ISSUE TITLE
project build status in gitea ui

### ISSUE THREAD

**Issue by celan** [aplanas/openSUSE-git#46]
Build the equivalent of a searchable/filterable build status overview of a project git in the gitea webui.

Requirements:
1) keep the build results in the package even after PR is merged
2) maintainers are missing  build results in gitea PR

**Comment by AdamMajer** [comment]
This can be done for projects as part of README.md in the project git. Status can be embedded as generated SVG.

For packages, this is much more tricky as README.md is static and it would refer to build results in devel project even when the package is embedded in different projects.

Also, while SVGs can contain Javascript, this is disabled when they are part of `<img>` element. SVGs references as `<object>` can have JS embedded, but these then can't be embedded in README.md. So there's a constraint on how much functionality you can embed in such an object.

Having said that, I think it's satisfactory to have devel project build status only in package build results as we build devel projects vs. targets

**Comment by JanZerebecki** [comment]
The status for a commit includes a detail link, which can usually be seen in the UI by hovering over the icon (for status: "pending", "success", "error" and "failure") after the commit message title, would also be a useful place to link to such a searchable/filterable status.

Is it sufficient to just link there to the one OBS provides?

Each status is identified by a context string (per commit/repo). While you could have one status for each build (combination of package/arch/etc), that is probably too much, as it is just shown in a list. You can give a short status in the description and update it from time to time, e.g. "OBS build status: 11 failure / 10 success / 987 building".

API: https://codeberg.org/api/swagger#/repository/repoCreateStatus https://docs.gitea.com/api#tag/repository/operation/repoCreateStatus
Example: https://codeberg.org/forgejo/forgejo/commit/c81e5fe123d6cecdd89b26a216922e2d72a66d2c

**Comment by JanZerebecki** [comment]
Example of the link one could use: https://build.opensuse.org/project/monitor/systemsmanagement:cockpit which is for the packages in https://src.opensuse.org/cockpit

**Comment by mmaslano** [comment]
@JanZerebecki do you see it as complete now? I can see on some projects pictures with builds and versions?

**Comment by JanZerebecki** [comment]
No, I think we should additionally set the build status for gitea, as it is directly shown on merge requests, not just provide an image to link in markdown files which we don't see without navigating to it.

**Comment by mmaslano** [comment]
Sadly unfinished project related to build results: https://hackweek.opensuse.org/25/projects/improve-build-results-for-gitea

**Comment by mmaslano** [comment]
Let's add build results in PR.

**Comment by fwag** [comment]
The staging bot is not yet adding the svg link as a comment in the project/package PR.

As requested by @gyr we need to implement an automated process to ensure that the submitter of a Pool PR is notified of a build failure that happens on the corresponding Forward PR.

**Comment by adrianschroeter** [comment]
what is the point of keeping the build results also after the merge?

The build state should always be "succeeded" or "excluded" if it got merged.

If you speak about the build artifacts like logfile or rpms then this is a scalability issue.

Why do you think you need that?

**Comment by adrianschroeter** [comment]
regarding the notification, the staging bot could add a success/fail message also into the package PR...

**Comment by fwag** [comment]
Note from Adrian: The bot must wait for the project to finish before sending a notification about the package. Even if the package build is successful, another package in the same staging PR can re-trigger it and cause it to fail later on.

**Comment by JanZerebecki** [comment]
(We should use @ the submitter in the failed comment.)

**Comment by mmaslano** [comment]
It already creates a comment with failed/succeed and a link to the OBS project. Probably good for now.
A link to buildresults webservice needs still to be added.

**Comment by fwag** [comment]
@bmwiedemann 
We still need to add a link to the SVG created by [br.opensuse.org](http://br.opensuse.org/) and [br.suse.de](http://br.suse.de/) in the staging bot. The build results server needs to be configured through an env variable.

In the project PR, the staging-bot shall add a comment with `![Build results](https://BR_API_URL/PR_PROJECT)`.
In the package PR, the staging-bot shall add a comment with `![Build results](https://BR_API_URL/PR_PROJECT/PACKAGE_NAME)`.

**Comment by bmwiedemann** [comment]
![test pkg Build results link](https://br.opensuse.org/status/GNOME:Factory/abiword)
-- 
![test prj Build results link](https://br.opensuse.org/status/GNOME:Factory)

**Comment by bmwiedemann** [comment]
Test-links look better on https://src.opensuse.org/git-workflow/autogits/pulls/199#issuecomment-114323 and are actual links, too.

https://src.opensuse.org/git-workflow/autogits/pulls/202 should solve this issue.

**Comment by mmaslano** [comment]
@adrianschroeter will review and deploy

**Comment by Ubayed-Bin-Sufian** [comment]
Hi! I have gone through the discussions and have taken some inspiration. 

right now, http://br.opensuse.org/ displays 404 error. so I attempted to implement the ideas and improve it in this PR. https://src.opensuse.org/git-workflow/autogits/pulls/192

The PR introduces a small frontend with: 
1. A preview of the build status with a loading indicator
2. A copyable generated link to the build status
3. A copyable markdown link for easy embedding

I would really appreciate feedback whether i am going in the right direction or i should adjust the approach.

**Comment by AdamMajer** [comment]
@Ubayed-Bin-Sufian thanks. I think we have 2 or 3 PRs open for this feature and no one managed to review these yet. Sorry that we can only merge one.. I'll try to find some time for this landing page.

**Comment by bmwiedemann** [comment]
That is already tracked in https://src.opensuse.org/git-workflow/autogits/issues/114, so can we close this issue now that https://src.opensuse.org/git-workflow/autogits/pulls/202 is merged?
